plugins/catalog-backend/src/ingestion/processors/AzureApiReaderProcessor.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { LocationSpec } from '@backstage/catalog-model';
+import fetch, { RequestInit, HeadersInit } from 'node-fetch';
+import * as result from './results';
+import { LocationProcessor, LocationProcessorEmit } from './types';
+
+export class AzureApiReaderProcessor implements LocationProcessor {
+  private privateToken: string = process.env.AZURE_PRIVATE_TOKEN || '';
```

Comment:
In the best of worlds, I would want this secret read out from our [config system](https://backstage.io/docs/conf/reading). Then, this class probably would just take the token as an optional string constructor parameter. I realise that the init of these classes is a bit dug down into the layers now, so it may be some work to propagate config to them. So for now, let's leave it like this and we'll look into getting the backend wiring more clearly configurable later.